### DISCUSSION TITLE
Dynamic data in pathless layouts

### DISCUSSION THREAD

**Discussion by colinramsay** [remix-run/remix#4300] [Q&A] [answered]
Given the following routes:

```
__pages.tsx
__pages/news.tsx
__pages/about.tsx
__pages/faq.tsx
```

I'd like `__pages.tsx` to load page data and populate header, hero image, etc, and then the sub-pages have their own individual content/formatting. 

My issue is that if I navigate to "news" and then go to "about", the content shown by `pages.tsx` doesn't change. I'm guessing that's because Remix doesn't think that's actually changing because it's not the "leaf" of the routes.

I'm struggling a little with this as it doesn't seem like I'm trying to do something too unreasonable so if anyone can sanity check this I'd appreciate it :smile:

**Comment by sergiodxa** [comment, marked as answer]
When only child routes change, Remix doesn’t call the loader of the parents route because the data is already there, because your __pages route is the parent of the other routes it’s doing this and avoiding calling it.


If you can to change the UI of __pages based on the child routes, you could use the useLocation hook to get the location object or use the useMatches hook to access the data from the current child route in the parent route and change the UI based on that.

**Reply by sergiodxa** [reply]
Also, if the data and UI of __pages change based on the child routes and it’s always different, you may not need to have that route at all and instead put the code in each route.

**Reply by colinramsay** [reply]
I ended up using `useMatches()`. Thanks for the speedy reply and advice, much appreciated!
